nextcord/embeds.py

```diff
@@ -427,17 +456,79 @@ def set_image(self: E, url: MaybeEmpty[Any]) -> E:
         -----------
         url: :class:`str`
             The source URL for the image. Only HTTP(S) is supported.
+        file: Optional[File]
+            A file to use for the image.
+
+            .. versionadded:: 2.0
         """
+        from . import InvalidArgument
 
         if url is EmptyEmbed:
             try:
                 del self._image
             except AttributeError:
                 pass
+
+        if file is EmptyEmbed:
+            try:
+                del self._local_files["image"]
+            except:
+                pass
+
+        elif url is not EmptyEmbed and file is not EmptyEmbed:
+            raise InvalidArgument("Can't pass both url and file.")
+
+        elif file is not EmptyEmbed:
+            self._local_files["image"] = file
+            self._image = {"url": f"attachment://{file.filename}"}
+
         else:
-            self._image = {
-                "url": str(url),
-            }
+            self._image = {"url": str(url)}
+
+        return self
+
+    def set_author(
+        self: E,
+        *,
+        name: Any,
+        url: MaybeEmpty[Any] = EmptyEmbed,
```

Comment:
shouldnt this be `MaybeEmpty[str]` as urls are strings

Reply:
Outside the scope of this pr. Just accidentally moved things which is why its green